pkg/amqp-bunny/AmqpProducer.php

```diff
@@ -50,12 +55,38 @@ class AmqpProducer implements InteropAmqpProducer, DelayStrategyAware
      */
     private $context;
 
+    /**
+     * @var bool
+     */
+    private $confirmModeEnabled = false;
+
+    /**
+     * @var float|null
+     */
+    private $timeout;
+
+    /**
+     * @var bool
+     */
+    private $ackCallbackAdded = false;
+
+    /**
+     * @var int|null
+     */
+    private $pendingDeliveryTag;
+
     public function __construct(Channel $channel, AmqpContext $context)
     {
         $this->channel = $channel;
         $this->context = $context;
     }
 
+    public function enableConfirmMode(float $timeout)
```

Comment:
This method must be a part of interface, say `RabbitmqPublisherConfirmsMode`. It should define a method that enable\disable, the mode as well as a method to wait for the result. 

The interface should be implemented on context, not a producer.

The interface should be in amqp-tools for now.